### PR TITLE
Feat/add Coin98 wallet

### PR DESCRIPTION
**Pull request by khiemsoft**
### Checklist
- [x] Increment the version field in `package.json` of the package you have made changes in following [semantic versioning](https://semver.org/) and using alpha release tagging
- [x] Check the box that allows repo maintainers to update this PR
- [x] Test locally to make sure this feature/fix works
- [x] Run `yarn check-all` to confirm there are not any associated errors
- [x] Confirm this PR passes Circle CI checks
- [x] Add or update relevant information in the documentation

### Docs Checklist
- [x] Include a screenshot of any changes ([see docs README on running locally](https://github.com/blocknative/web3-onboard/blob/develop/docs/README.md))
- [x] Add/update the appropriate package README (if applicable)
- [x] Add/update the related module in the [docs demo](https://github.com/blocknative/web3-onboard/blob/develop/docs/src/lib/services/onboard.js) (if applicable)
- [x] Add/update the related package in the `docs/package.json` file (if applicable)

### If this PR includes changes to add an injected wallet or SDK wallet module: 
Please complete the following using the internal demo package.
To run this demo use the command `yarn && yarn dev` to get the project running at `http://localhost:8080/`

#### Tests with demo app (injected)
- [ ] send transaction
- [ ] switch chains
- [ ] sign message
- [ ] sign typed message
- [ ] disconnect

#### Tests with demo app (SDK)
- [ ] send transaction
- [ ] switch chains
- [ ] sign message
- [ ] sign typed message
- [ ] disconnect

### CODE REVIEW

**Comment by leightkt** [comment]
@khiemsoft this is looking good- can you give me access to the repo so I can test locally?

**Comment by khiemsoft** [comment]
> @khiemsoft this is looking good- can you give me access to the repo so I can test locally?

It's done. Could you please check? Thanks for your review.

**Comment by leightkt** [comment]
@khiemsoft You need to update the version on the package.json for  packages/injected/package.json to 2.10.3-alpha.2
and then updated the "@web3-onboard/injected-wallets" version in packages/demo/package.json to match in order to be able to follow the instructions for starting the demo for testing- I was unable to connect to coin98 wallet on desktop or mobile. Please run through all tests listed in the checklist above (send transaction, sign message, etc) and let me know when you have completed this. Thanks!

Screenshot 2023-07-15 at 11 50 04 AM

**Comment by khiemsoft** [comment]
> @khiemsoft You need to update the version on the package.json for packages/injected/package.json to 2.10.3-alpha.2 and then updated the "@web3-onboard/injected-wallets" version in packages/demo/package.json to match in order to be able to follow the instructions for starting the demo for testing- I was unable to connect to coin98 wallet on desktop or mobile. Please run through all tests listed in the checklist above (send transaction, sign message, etc) and let me know when you have completed this. Thanks!
> 
> Screenshot 2023-07-15 at 11 50 04 AM

Thanks for your review. All changes are updated and have already been tested. Could you please check?

**Comment by leightkt** [comment]
@khiemsoft Thanks for updating! I tested locally and was able to connect, disconnect, and sign a message.
I could not sign a typed message or switch chains.

**Comment by khiemsoft** [comment]
> @khiemsoft Thanks for updating! I tested locally and was able to connect, disconnect, and sign a message. I could not sign a typed message or switch chains.

Hi, this is a bug in the custom network in the Coin98 Wallet Extension. We have just updated to version 7.2.7. Could you please update it and test it again? Thanks for your review.

**Comment by leightkt** [comment]
@khiemsoft I can now successfully connect, disconnect, sign a message, sign a typed message, and switch chains. 
I cannot send a transaction (attempted on Goerli)

File: packages/injected/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@web3-onboard/injected-wallets",
-  "version": "2.10.3",
+  "version": "2.10.3-alpha.2",
```
Comment (leightkt) [inline packages/injected/package.json:3]:
```suggestion
  "version": "2.10.4-alpha.1",
```

**Comment by khiemsoft** [comment]
> @khiemsoft I can now successfully connect, disconnect, sign a message, sign a typed message, and switch chains. I cannot send a transaction (attempted on Goerli) Screenshot 2023-07-24 at 11 34 12 AM

@leightkt  Hi, could you please check again? My transaction is still fine. Here is my example video
https://github.com/blocknative/web3-onboard/assets/87562184/3332961b-58bc-4131-966a-e97580f5dede

**Comment by leightkt** [comment]
@khiemsoft I'm still getting the same error. I wonder if it has something to do with how I've set up my wallet? I've tried to transfer Goerli ETH into my coin98 from my metamask wallet twice - both transactions were successful on etherscan, but my coin98 wallet is still telling me I have a 0 balance: 
Screenshot 2023-07-28 at 12 08 52 PM
Screenshot 2023-07-28 at 12 09 08 PM

**Comment by khiemsoft** [comment]
> @khiemsoft I'm still getting the same error. I wonder if it has something to do with how I've set up my wallet? I've tried to transfer Goerli ETH into my coin98 from my metamask wallet twice - both transactions were successful on etherscan, but my coin98 wallet is still telling me I have a 0 balance: Screenshot 2023-07-28 at 12 08 52 PM Screenshot 2023-07-28 at 12 09 08 PM

The balance of the testnet main token will not be displayed on this main display screen. You need to scroll down, and you will see a custom token like this. @leightkt 
image

**Comment by leightkt** [comment]
@khiemsoft Ok thanks- I'm still getting the same error as above when trying to send a tx

**Comment by khiemsoft** [comment]
> @khiemsoft Ok thanks- I'm still getting the same error as above when trying to send a tx

Hi, I just found that your key when added to the Infura Goerli custom network is an error.

Could you please add it from the Coin98 Wallet custom network or try another key in Infura? Thanks for your review.

**Comment by leightkt** [comment]
@khiemsoft I tried a different rpcURL and I tried adding from the wallet directly and I still get the same error

**Comment by khiemsoft** [comment]
> @khiemsoft I tried a different rpcURL and I tried adding from the wallet directly and I still get the same error

Hi, this is our re-test video. This is how to reproduce your bug and how to change this key. Thanks for your review. @leightkt 

https://github.com/blocknative/web3-onboard/assets/87562184/cd5dac50-19cc-460f-b3a8-430f8ef8e10b

**Comment by khiemsoft** [comment]
> @khiemsoft I had to go through a few extra steps to reactivate a goerli wallet after removing the network but I got there and was able to send a transaction.
> 
> Please pull in the most recent changes from develop into your branch and resolve any conflicts and this will be good to go.

Conflicts fixed. Thanks for your review. @leightkt

File: packages/injected/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@web3-onboard/injected-wallets",
-  "version": "2.10.4",
+  "version": "2.10.4-alpha.1",
```
Comment (leightkt) [inline packages/injected/package.json:3]:
```suggestion
  "version": "2.10.5-alpha.1",
```

**Comment by leightkt** [comment]
@khiemsoft left a comment to update the injected package version
please also update package/demo/package.json "@web3-onboard/injected-wallets" to "^2.10.5-alpha.1"

**Comment by khiemsoft** [comment]
> @khiemsoft left a comment to update the injected package version please also update package/demo/package.json "@web3-onboard/injected-wallets" to "^2.10.5-alpha.1"

Package version updated Thanks for your review.

**Comment by leightkt** [comment]
@khiemsoft thanks so much for getting this across the finish line! It will go out in our next release!

**Pull request merged**